python-lib/nlp_utils.py

```diff
@@ -3,16 +3,40 @@
 import unicodedata
 
 
-def get_keyword(text: AnyStr, normalize_case: bool) -> AnyStr:
+def normalize_case(text: AnyStr, lowercase: bool) -> AnyStr:
     """Return text in its wanted-case form"""
-    return text.lower() if normalize_case else text
+    return text.lower() if lowercase else text
 
 
-def get_sentence(span: Span, normalize_case: bool) -> Union[Span, Doc]:
-    """Return Span object as a Doc if case_insensitive is set to True"""
-    return span if normalize_case else span.as_doc()
+def get_token_attribute(lemmatize: bool) -> AnyStr:
+    """Return the attribute to pass to spaCy Matcher"""
+    return "LEMMA" if lemmatize else "TEXT"
 
 
-def normalize_text(texts: List[AnyStr]) -> List[AnyStr]:
+def lemmatize(text: Doc, lowercase: bool) -> AnyStr:
+    """Lemmatize text. The lemma is lowercased if 'lowercase' is True"""
+    text = " ".join([word.lemma_ for word in text])
+    return normalize_case(text, lowercase)
```

Comment:
Better to separate concerns: function called lemmatize should just lemmatize, not lemmatize and lowercase. Either rename it to "lemmative_and_lowercase" or move the lowercasing to the `normalize` function below

Reply:
d807407